### ISSUE TITLE
Transfer report error when value is empty

### ISSUE THREAD

**Issue by jiangtj** [element-plus/element-plus#885]
Below is my code
```html
  <el-transfer 
    class="dialog-el-transfer" 
    v-model="transferVal" 
    :props="{ key: 'id', label: 'name' }" 
    :data="roles" 
    :titles="['Source', 'Target']"
  />
```
```js
  data() {
    return {
      transferVal: [],
      roles: [ {
        "id" : 1,
        "name" : "admin"
      }, {
        "id" : 2,
        "name" : "test"
      }]
    }
  }
```

When I select the item on the left and move it to the right, it will report the following error

```
Uncaught (in promise) TypeError: Cannot read property '0' of null
    at patchBlockChildren (runtime-core.esm-bundler.js?5c40:3996)
   // ...
```

This error occurs only if there is no data on the left or right panel

**Comment by Ryan2128** [comment]
Are you using the latest version?

It works well in latest version.

[demo](https://codesandbox.io/s/practical-sanderson-0p31h?file=/src/App.vue)

**Comment by jiangtj** [comment]
```json
"dependencies": {
    "axios": "^0.21.0",
    "core-js": "^3.8.1",
    "element-plus": "^1.0.1-beta.5",
    "ismobilejs": "^1.1.1",
    "qs": "^6.9.4",
    "vue": "^3.0.0",
    "vue-router": "^4.0.0-0",
    "vuex": "^4.0.0-0"
  },
  "devDependencies": {
    "@vue/cli-plugin-babel": "~4.5.0",
    "@vue/cli-plugin-eslint": "~4.5.0",
    "@vue/cli-plugin-router": "~4.5.0",
    "@vue/cli-plugin-vuex": "~4.5.0",
    "@vue/cli-service": "~4.5.0",
    "@vue/compiler-sfc": "^3.0.4",
    "babel-eslint": "^10.1.0",
    "eslint": "^7.15.0",
    "eslint-plugin-vue": "^7.2.0",
    "sass": "^1.30.0",
    "sass-loader": "^10.1.0",
    "stylelint": "^13.8.0",
    "stylelint-config-standard": "^20.0.0"
  }
```

This is my dependency, and `beta.5` is already the latest version released yesterday

**Comment by bastarder** [comment]
i test this case, it works on `vue: 3.0.3` but not `vue: 3.0.4`

**Comment by Ryan2128** [comment]
I reproduced it, I am going to check the reason.

**Comment by bastarder** [comment]
> I reproduced it, I am going to check the reason.

i test this case, it work on 'Vue: 3.0.3' but can not work on 'Vue: 3.0.4'

**Comment by JeremyWuuuuu** [comment]
> > I reproduced it, I am going to check the reason.
> 
> i test this case, it work on 'Vue: 3.0.3' but can not work on 'Vue: 3.0.4'

Based on the fact that `Vue` itself has bug, this is not a bug from us, its a bug from `Vue`, check this out: https://github.com/vuejs/vue-next/commit/c59897c7b0dbd82b5bbf3fbca945c0639ac37fb8

**Comment by JeremyWuuuuu** [comment]
Unless `Vue` release the fix in `3.0.5` we cannot help you with this. Downgrade you version of `Vue` to `3.0.3`, it will work.
